pkg/kube/controllers/extendedjob/errand_controller.go

```diff
@@ -153,14 +153,16 @@ func AddErrand(ctx context.Context, config *config.Config, mgr manager.Manager)
 	}
 
 	// Watch secrets referenced by resource ExtendedJob
-	// trigger auto errand if UpdateOnConfigChange=true and config data changed
+	// trigger auto errand if UpdateOnConfigChange=true and config data changed.
+	// This includes changes to any version of a referenced 'versioned secret'.
```

Comment:
They should be immutable..
If possible, we should ignore "changes" to any secret that is versioned.
We should only care about creation.